internal/clients/http/sources/sources_client.go

```diff
@@ -186,14 +185,23 @@ func (c *sourcesClient) GetAuthentication(ctx context.Context, sourceId string)
 		return nil, fmt.Errorf("get source authentication call: %w", err)
 	}
 
-	if len(*resp.JSON200.Data) != 0 {
-		auth := (*resp.JSON200.Data)[0]
-		authentication := clients.NewAuthenticationFromSourceAuthType(ctx, *auth.Username, string(*auth.Authtype), *auth.ResourceId)
-		return authentication, nil
-	} else {
-		logger.Trace().Msgf("Source does not have provisioning authentications of type application")
-		return nil, clients.MissingProvisioningSources
+	// Filter authentications to include only auth where resource_type == "Application". We do this because
+	// Sources API currently does not provide a good server-side filtering.
```

Comment:
Maybe drop a note that we expect this endpoint to never return more than dozen of entries. Just for the record, its three if you enable launch and image builder via super key:

```
$ curl-stage GET "/sources/v3.1/sources/415043/authentications" | jq
{
  "data": [
    {
      "id": "353870",
      "authtype": "access_key_secret_key",
      "username": "XXX",
      "availability_status": "in_progress",
      "resource_type": "Source",
      "resource_id": "415043"
    },
    {
      "id": "353872",
      "authtype": "cloud-meter-arn",
      "username": "arn:aws:iam::XXX:role/redhat-cloud-meter-role-2b61b90877ea2d4f",
      "availability_status": "in_progress",
      "resource_type": "Application",
      "resource_id": "353442"
    },
    {
      "id": "353871",
      "authtype": "provisioning-arn",
      "username": "arn:aws:iam::XXX:role/redhat-provisioning-role-0dbb45280bca3ab0",
      "availability_status": "in_progress",
      "resource_type": "Application",
      "resource_id": "353441"
    }
  ],
  "meta": {
    "count": 3,
    "limit": 100,
    "offset": 0
  },
  "links": {
    "first": "/api/sources/v3.1/sources/415043/authentications?limit=100&offset=0",
    "last": "/api/sources/v3.1/sources/415043/authentications?limit=100&offset=100"
  }
}
```